codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -467,6 +436,96 @@ public void settingsWriteShouldNotUpdateSettings() throws Exception {
         verify(writer).writeSettings(eq(ctx), eq(settings), eq(promise));
     }
 
+    @Test
+    public void dataWriteShouldCreateHalfClosedStream() {
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                FlowControlled flowControlled = (FlowControlled) invocationOnMock.getArguments()[2];
+                flowControlled.write(Integer.MAX_VALUE);
+                flowControlled.writeComplete();
+                return null;
+            }
+        }).when(remoteFlow).sendFlowControlled(eq(ctx), eq(stream), any(FlowControlled.class));
+        ByteBuf data = dummyData();
+        encoder.writeData(ctx, STREAM_ID, data.retain(), 0, true, promise);
+        verify(remoteFlow).sendFlowControlled(eq(ctx), eq(stream), any(FlowControlled.class));
+        verify(lifecycleManager).closeLocalSide(stream, promise);
+        assertEquals(data.toString(UTF_8), writtenData.get(0));
+        data.touch();
+    }
+
+    @Test
+    public void dataWriteShouldHalfCloseStream() throws Exception {
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                FlowControlled flowControlled = (FlowControlled) invocationOnMock.getArguments()[2];
+                flowControlled.write(Integer.MAX_VALUE);
+                flowControlled.writeComplete();
+                return null;
+            }
+        }).when(remoteFlow).sendFlowControlled(eq(ctx), eq(stream), payloadCaptor.capture());
+        reset(future);
+        final ByteBuf data = dummyData();
+        encoder.writeData(ctx, STREAM_ID, data, 0, true, promise);
+        assertEquals(1, payloadCaptor.getAllValues().size());
+        verify(lifecycleManager).closeLocalSide(eq(stream), eq(promise));
+        assertEquals(0, data.refCnt());
+    }
+
+    @Test
+    public void headersWriteShouldHalfCloseStream() throws Exception {
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                FlowControlled flowControlled = (FlowControlled) invocationOnMock.getArguments()[2];
+                flowControlled.write(Integer.MAX_VALUE);
+                flowControlled.writeComplete();
+                return null;
+            }
+        }).when(remoteFlow).sendFlowControlled(eq(ctx), eq(stream), payloadCaptor.capture());
+        int streamId = 5;
```

Comment:
This may have just been carry over from another test but do we need this?  Typically we only need to define another stream id in the test if there are multiple streams...otherwise just use `STREAM_ID`.  We may be able to remove a few mock calls if we use `STREAM_ID` too.